waku/waku_api/rest/relay/handlers.nim

```diff
@@ -135,16 +135,8 @@ proc installRelayApiHandlers*(router: var RestRouter, node: WakuNode, cache: Mes
       if not success:
         return RestApiResponse.internalServerError("Failed to publish: error appending RLN proof to message")
 
-      # validate the message before sending it
-      let result = node.wakuRlnRelay.validateMessageAndUpdateLog(message)
-      if result == MessageValidationResult.Invalid:
-        return RestApiResponse.internalServerError("Failed to publish: invalid RLN proof")
-      elif result == MessageValidationResult.Spam:
-        return RestApiResponse.badRequest("Failed to publish: limit exceeded, try again later")
-      elif result == MessageValidationResult.Valid:
-        debug "RLN proof validated successfully", pubSubTopic=pubSubTopic
-      else:
-        return RestApiResponse.internalServerError("Failed to publish: unknown RLN proof validation result")
+      (await node.wakuRelay.validateMessage(pubsubTopic, message)).isOkOr:
```

Comment:
imho you should check also the result. Its not only about the result being Ok, its also about being `Valid`. If this returns `Spam` or `Invalid` we should error.  (see code that was removed)

Reply:
`validateMessage()` returns `ok()` if the message is validated by all the subscribed validators, and `err("error message")` if the message is not validated. The result doesn't contain anything, its return type is `Result[void, string]`.

The generic validators can't return the results `Spam` or `Invalid`, their return type is `ValidationResult` which is used by libp2p (as the validators are also passed to libp2p layer), so it does not have the RLN specific return values.

This is one of the big tradeoffs of this approach using generic validators instead of hardcoded checks in the REST handlers. Wrote about this on the following comment: https://github.com/waku-org/nwaku/pull/2373/files#r1467933272

Not sure what's the right thing to do. But if we want to run our registered validators in the REST handler to notify the user if there's a validation error, and the RLN validator is part of our validator list (so it will be run), then the best I think we can do is logging the specific result and return the values that they allow. LMK if I'm missing something or if doesn't seem worth it

Reply:
right, apologies I overlooked it.